src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -44,7 +44,7 @@ struct ImagePayloadBase : public PayloadBase {
   std::string peer_mirror_uuid;
   std::string peer_image_id;
 
-  ImagePayloadBase() : PayloadBase() {
+  ImagePayloadBase() : PayloadBase(), global_image_id{""}, peer_mirror_uuid{""}, peer_image_id{""} {
```

Comment:
Hmm -- I really don't want to add workarounds for apparent GCC bugs. Given that GCC 7.1.x was released last month, I'd be inclined to just report this upstream to them instead.